src/Microsoft.ApiDesignGuidelines.Analyzers/Core/IdentifiersShouldNotMatchKeywords.cs

```diff
@@ -1,69 +1,621 @@
+
+        // Format the symbol name in a way consistent with FxCop's display for this rule.
+        private static string FormatSymbolName(ISymbol symbol)
+        {
+            return symbol.ToDisplayString(
+                // This format omits the namespace.
+                SymbolDisplayFormat.CSharpShortErrorMessageFormat
+                    // Turn off the EscapeKeywordIdentifiers flag (which is on by default), so that
+                    // a method named "@for" is displayed as "for".
+                    // Turn on the UseSpecialTypes flat (which is off by default), so that parameter
+                    // names of "special" types such as Int32 are displayed as their language alias,
+                    // such as int for C# and Integer for VB.
+                    .WithMiscellaneousOptions(SymbolDisplayMiscellaneousOptions.UseSpecialTypes));
         }
+
+        private ImmutableHashSet<string> s_caseSensitiveKeywords = new string[]
+        {
+            // C#
+            "abstract",
+            "as",
+            "base",
+            "bool",
+            "break",
+            "byte",
+            "case",
+            "catch",
+            "char",
+            "checked",
+            "class",
+            "const",
+            "continue",
+            "decimal",
+            "default",
+            "delegate",
+            "do",
+            "double",
+            "else",
+            "enum",
+            "event",
+            "explicit",
+            "extern",
+            "false",
+            "finally",
+            "fixed",
+            "float",
+            "for",
+            "foreach",
+            "goto",
+            "if",
+            "implicit",
+            "in",
+            "int",
+            "interface",
+            "internal",
+            "is",
+            "lock",
+            "long",
+            "namespace",
+            "new",
+            "null",
+            "object",
+            "operator",
+            "out",
+            "override",
+            "params",
+            "private",
+            "protected",
+            "public",
+            "readonly",
+            "ref",
+            "return",
+            "sbyte",
+            "sealed",
+            "short",
+            "sizeof",
+            "static",
+            "string",
+            "struct",
+            "switch",
+            "this",
+            "throw",
+            "true",
+            "try",
+            "typeof",
+            "uint",
+            "ulong",
+            "unchecked",
+            "unsafe",
+            "ushort",
+            "using",
+            "virtual",
+            "void",
+            "volatile",
+            "while",
+            // Listed as a keywords in Microsoft.CodeAnalysis.CSharp.SyntaxKind, but
+            // omitted, at least for now, for compatibility with FxCop:
+            //"arglist", (Shouldn't this be "__arglist"?)
+            //"makeref",
+            //"reftype",
+            //"refvalue",
```

Comment:
It should be __arglist, __makeref, __reftype, and __refvalue. I'd changed them (but leave commented for compat) and remove the question. 

These keywords are not documented in the language spec IIRC. Not checking them isn't too bad as we will also flag the underscores.